internal/kafka/internal/services/kafka_instance_types.go

```diff
@@ -0,0 +1,89 @@
+}
+
+type supportedKafkaInstanceTypesService struct {
+	providerConfig *config.ProviderConfig
+	kafkaConfig    *config.KafkaConfig
+}
+
+func NewSupportedKafkaInstanceTypesService(providerConfig *config.ProviderConfig, kafkaConfig *config.KafkaConfig) SupportedKafkaInstanceTypesService {
+	return &supportedKafkaInstanceTypesService{
+		providerConfig: providerConfig,
+		kafkaConfig:    kafkaConfig,
+	}
+}
+
+func (t supportedKafkaInstanceTypesService) GetSupportedKafkaInstanceTypesByRegion(providerId string, regionId string) ([]api.SupportedKafkaInstanceType, *errors.ServiceError) {
+	instanceTypeList := []api.SupportedKafkaInstanceType{}
+
+	provider, providerFound := t.providerConfig.ProvidersConfig.SupportedProviders.GetByName(providerId)
+	if !providerFound {
+		return nil, errors.ProviderNotSupported(fmt.Sprintf("cloud provider '%s' is unsupported", providerId))
+	}
+
+	region, regionFound := provider.Regions.GetByName(regionId)
+	if !regionFound {
+		return nil, errors.RegionNotSupported(fmt.Sprintf("cloud region '%s' is unsupported", regionId))
+	}
+
+	for k := range region.SupportedInstanceTypes {
+		instanceType, err := t.kafkaConfig.SupportedInstanceTypes.Configuration.GetKafkaInstanceTypeByID(k)
+		if err != nil {
+			return nil, errors.InstanceTypeNotSupported(fmt.Sprintf("instance type '%s' is unsupported", k))
+		}
+		sizes := instanceType.Sizes
+		supportedSizesList := []api.SupportedKafkaSize{}
+		for _, size := range sizes {
+			supportedSizesList = append(supportedSizesList, api.SupportedKafkaSize{
+				Id: size.Id,
+				IngressThroughputPerSec: api.SupportedKafkaSizeBytesValueItem{
+					Bytes: convertStringToValue(size.IngressThroughputPerSec),
+				},
+				EgressThroughputPerSec: api.SupportedKafkaSizeBytesValueItem{
+					Bytes: convertStringToValue(size.EgressThroughputPerSec),
+				},
+				TotalMaxConnections: int32(size.TotalMaxConnections),
+				MaxDataRetentionSize: api.SupportedKafkaSizeBytesValueItem{
+					Bytes: convertStringToValue(size.MaxDataRetentionSize),
+				},
+				MaxPartitions:               int32(size.MaxPartitions),
+				MaxDataRetentionPeriod:      size.MaxDataRetentionPeriod,
+				MaxConnectionAttemptsPerSec: int32(size.MaxConnectionAttemptsPerSec),
+				QuotaConsumed:               int32(size.QuotaConsumed),
+				QuotaType:                   size.QuotaType,
+				CapacityConsumed:            int32(size.CapacityConsumed),
+			})
+		}
+		instanceTypeList = append(instanceTypeList, api.SupportedKafkaInstanceType{
+			Id:                  k,
+			SupportedKafkaSizes: supportedSizesList,
+		})
+	}
+	sort.Slice(instanceTypeList, func(i, j int) bool {
+		return instanceTypeList[i].Id < instanceTypeList[j].Id
+	})
+
+	return instanceTypeList, nil
+}
+
+//convertStringToValue converts string value used to create a resource.Quantity type to float32
+//input string values have already been validated
+func convertStringToValue(input string) float32 {
```

Comment:
I think this is generic enough, this could probably go into the [shared](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/tree/main/pkg/shared) pkg? 

Let's change the function name to be specific to what it does, maybe something like `ConvertStrQuantityToFloat()`?

Reply:
If float32 isn't required, maybe we could use the util function from k8s for converting this to float64 [AsApproximateFloat64](https://pkg.go.dev/k8s.io/apimachinery@v0.21.2/pkg/api/resource#Quantity.AsApproximateFloat64)?

Reply:
What about creating a 'Quantity' type with a `ToFloat32` method?
That way we could use `Quantity` instead of `string` and code would be more readable
WDYT?

Reply:
Good idea about placing in the shared pkg. I'll use the AsApproximateFloat64() for the shared function. Thanks

Reply:
@ziccardi missed your comment earlier. Sounds good